app/javascript/components/domain-form/index.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { API } from '../../http_api';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import schema from './domain-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import handleFailure from '../../helpers/handle-failure';
+
+const getInitialValues = id => API.get(`/api/automate_domains/${id}?attributes=name,description,enabled,source`);
+
+const handleSubmit = (id, values, redirectUrl) => {
+  miqSparkleOn(); // miqRedirectBack turns it off
+  const action = id ? 'edit' : 'create';
+  const flashMessage = sprintf(__('Automate Domain "%s" was %s.'),
+    values.name,
+    id ? 'saved' : 'added');
+
+  API.post(`/api/automate_domains/${id || ''}`, {
+    action, ...values,
+  }, { skipErrors: [400] })
+    .then(() => miqRedirectBack(flashMessage, 'success', redirectUrl))
+    .catch(handleFailure);
+};
+
+const NamespaceForm = ({ id }) => {
+  const [state, setState] = useState({ name: '', description: '', enabled: true, readonly: false, loaded: false });
+  const redirectUrl = '/miq_ae_class/explorer';
+  const cancelMessage = sprintf(__('%s of Domain "%s" was cancelled by user.'),
```

Comment:
same here, "Add of Domain foo" is not even english ;). 

2 actions = 2 strings please